### PR TITLE
OSDOCS-2279: Document managementState API for DNS

### PR DESCRIPTION
**Pull request by bmcelvee**
https://issues.redhat.com/browse/OSDOCS-2279

Preview: https://deploy-preview-36197--osdocs.netlify.app/openshift-enterprise/latest/networking/dns-operator.html#nw-dns-operator-managementState_dns-operator

### CODE REVIEW

**Comment by bmcelvee** [comment]
@rfredette would you mind giving this PR a review, please? Thank you!

**Comment by bmcelvee** [comment]
@lihongan would you mind reviewing this PR, please? Thank you!

**Comment by rfredette** [comment]
lgtm :+1:

File: modules/nw-dns-operator-managementState.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+* You are a cluster administrator and have reported an issue with CoreDNS, but need to apply a workaround until the issue is fixed. You can set the `managementState` field of the DNS Operator to `Unmanaged` to apply the workaround.
+
+.Procedure
+
+* Change `managementState` DNS Operator:
++
+[source,terminal]
+----
+$ $ oc patch configs.dns.operator.openshift.io cluster --type merge --patch '{"spec":{"managementState":"Unmanaged"}}'
```
Comment (sagidlow) [inline modules/nw-dns-operator-managementState.adoc:22]:
```suggestion
$ oc patch configs.dns.operator.openshift.io cluster --type merge --patch '{"spec":{"managementState":"Unmanaged"}}'
```

Extra $

File: modules/nw-dns-operator-managementState.adoc [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Module included in the following assemblies:
+//
+// * networking/dns-operator.adoc
+
+[id="nw-dns-operator-managementState_{context}"]
+= Changing the DNS Operator managementState
+
+DNS manages the CoreDNS component to provide a name resolution service for pods and services in the cluster. The `managementState` of the DNS Operator is set to `Managed` by default, which means that the DNS Operator is actively managing its resources. You can change it to `Unmanaged`, which means the DNS Operator is not managing its resources.
+
+Use cases for changing the DNS Operator `managementState` include:
```
Comment (sagidlow) [inline modules/nw-dns-operator-managementState.adoc:10]: Suggestion below based on IBM style guide for lead-in wording. It's a bit more straightforward for translation purposes. 

```suggestion
The following are use cases for changing the DNS Operator `managementState`:
```

**Review by sagidlow — COMMENTED** [review]
Two small comments. Otherwise, looks great!

File: modules/nw-dns-operator-managementState.adoc (hunk 1 above)
Comment (bmcelvee) [reply modules/nw-dns-operator-managementState.adoc:22]: Oh no, silly mistake. Thank you!

File: modules/nw-dns-operator-managementState.adoc [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+* You are a cluster administrator and have reported an issue with CoreDNS, but need to apply a workaround until the issue is fixed. You can set the `managementState` field of the DNS Operator to `Unmanaged` to apply the workaround.
+
+.Procedure
+
+* Change `managementState` DNS Operator:
++
+[source,terminal]
+----
+$ oc patch configs.dns.operator.openshift.io cluster --type merge --patch '{"spec":{"managementState":"Unmanaged"}}'
```
Comment (lihongan) [inline modules/nw-dns-operator-managementState.adoc:22]: I cannot find the resource `configs.dns.operator.openshift.io` and seems the command doesn't work.
```
$ oc patch configs.dns.operator.openshift.io cluster --type merge --patch '{"spec":{"managementState":"Unmanaged"}}'
error: the server doesn't have a resource type "configs"
```
@Miciah could you help confirm

Comment (lihongan) [reply modules/nw-dns-operator-managementState.adoc:22]: I found one resource named `configs.operator.openshift.io` has the `.spec.managementState` but it looks not for DNS operator, see
```
$ oc get configs.operator.openshift.io cluster -ojson | jq '.spec'
{
  "logLevel": "Normal",
  "managementState": "Managed",
  "operatorLogLevel": "Normal"
}
```

Comment (lihongan) [reply modules/nw-dns-operator-managementState.adoc:22]: According to https://github.com/openshift/api/pull/888/files, the correct command should be
```
oc patch dns.operator.openshift.io default --type merge --patch '{"spec":{"managementState":"Unmanaged"}}'
```

Comment (bmcelvee) [reply modules/nw-dns-operator-managementState.adoc]: Thank you! PR updated.

**Comment by lihongan** [comment]
LGTM
Thank you @bmcelvee

**Pull request merged**